ethsnarks/mimc.py

```diff
@@ -0,0 +1,232 @@
+"""
+https://eprint.iacr.org/2016/492.pdf
+https://eprint.iacr.org/2016/542.pdf
+https://github.com/zcash/zcash/issues/2233
+https://keccak.team/files/SpongeIndifferentiability.pdf
+
+Where the field F_p is prime, it needs to be assured that
+the cubing in the round function creates a permutation.
```

Comment:
Well, it isn't strictly necessary for the Feistel variant. The [paper](https://eprint.iacr.org/2016/492.pdf) says in section 5.3:
> Remember that for MiMC-n/n, d has to satisfy the condition gcd(d, 2<sup>n</sup> - 1) = 1 in order to be a permutation, while in the case of MiMC-2n/n (that is, for Feistel Networks) this condition is not necessary.

There is insufficient evidence to say which e is more secure for a given number of rounds.